drc_cmis/cmis/utils.py

```diff
@@ -1,161 +1,361 @@
+            properties[prop_node.nodeName] = value
+
+    return properties
+
+
+# FIXME find a better way to do this
+def extract_content(soap_response_body: str) -> BytesIO:
+    xml_response = extract_xml_from_soap(soap_response_body)
+    extracted_data = extract_content_stream_properties_from_xml(xml_response)
+
+    # After the filename comes the file content
+    last_header = (
+        f"Content-Disposition: attachment;name=\"{extracted_data['filename']}\""
+    )
+    idx_content = soap_response_body.find(last_header) + len(last_header)
+    content_with_boundary = soap_response_body[idx_content:]
+    content = re.search("\r\n\r\n(.+?)\r\n--uuid:.+?--", content_with_boundary).group(1)
+
+    return BytesIO(content.encode())
+
+
+def make_soap_envelope(
+    cmis_action: str,
+    repository_id: Optional[str] = None,
+    properties: Optional[dict] = None,
+    statement: Optional[str] = None,
+    object_id: Optional[str] = None,
+    folder_id: Optional[str] = None,
+    content_id: Optional[str] = None,
+    major: Optional[str] = None,
+    checkin_comment: Optional[str] = None,
+) -> minidom.Document:
+
+    xml_doc = minidom.Document()
+
+    # Main soap entry element
+    entry_element = xml_doc.createElement("soapenv:Envelope")
+    entry_element.setAttribute(
+        "xmlns:soapenv", "http://schemas.xmlsoap.org/soap/envelope/"
+    )
+    entry_element.setAttribute(
+        "xmlns:ns", "http://docs.oasis-open.org/ns/cmis/messaging/200908/"
+    )
+    entry_element.setAttribute(
+        "xmlns:ns1", "http://docs.oasis-open.org/ns/cmis/core/200908/"
+    )
+    xml_doc.appendChild(entry_element)
+
+    # Creates the security header
+    header_element = xml_doc.createElement("soapenv:Header")
+    security_header = xml_doc.createElement("Security")
+    security_header.setAttribute(
+        "xmlns",
+        "http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd",
+    )
+
+    # Time stamp
+    time_stamp_tag = xml_doc.createElement("Timestamp")
+
+    created_tag = xml_doc.createElement("Created")
+    created_text = xml_doc.createTextNode(datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ"))
+    created_tag.appendChild(created_text)
+    time_stamp_tag.appendChild(created_tag)
+
+    expires_tag = xml_doc.createElement("Expires")
+    expires_text = xml_doc.createTextNode(
+        (datetime.now() + timedelta(1)).strftime("%Y-%m-%dT%H:%M:%SZ")
+    )
+    expires_tag.appendChild(expires_text)
+    time_stamp_tag.appendChild(expires_tag)
+
+    security_header.appendChild(time_stamp_tag)
+
+    # Username token
+    username_token_tag = xml_doc.createElement("UsernameToken")
+
+    username_tag = xml_doc.createElement("Username")
+    username_text = xml_doc.createTextNode("admin")
+    username_tag.appendChild(username_text)
+    username_token_tag.appendChild(username_tag)
+
+    password_tag = xml_doc.createElement("Password")
+    password_text = xml_doc.createTextNode("admin")
```

Comment:
Isn't this better to put this in settings? That way it is easier to override them when needed.

Reply:
These now use the values that are in the CMISConfig solo object